modeshape-jcr/src/main/java/org/modeshape/jcr/value/binary/Database.java

```diff
@@ -155,11 +159,29 @@ protected Database( Connection connection,
         }
         this.prefix = prefix == null ? null : prefix.trim();
         this.tableName = this.prefix != null && this.prefix.length() != 0 ? this.prefix + TABLE_NAME : TABLE_NAME;
+        this.maxExtractedTextLength = determineMaxExtractedTextLength(metaData);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Using max length for extracted text '{0}'", maxExtractedTextLength);
+        }
 
         initializeStatements();
         initializeStorage(connection);
     }
 
+    private int determineMaxExtractedTextLength( DatabaseMetaData metaData ) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = metaData.getColumns(null, null, null,  EXTRACTED_TEXT_COLUMN_NAME);
+            return resultSet.next() ? resultSet.getInt("COLUMN_SIZE") : DEFAULT_MAX_EXTRACTED_TEXT_LENGTH;
```

Comment:
This will find all of the columns with that name in any table in any schema in any database, right? And won't it be possible to return more than one result? After seeing this code, this seems pretty brittle.